src/test/java/org/folio/rest/impl/MockServer.java

```diff
@@ -626,25 +627,43 @@ private void handleGetInvoices(RoutingContext ctx) {
     } else if (queryParam.contains(ID_FOR_INTERNAL_SERVER_ERROR)) {
       serverResponse(ctx, 500, APPLICATION_JSON, Response.Status.INTERNAL_SERVER_ERROR.getReasonPhrase());
     } else if (queryParam.startsWith(QUERY_PARAM_START_WITH)) {
+      Matcher lineIdMatcher = Pattern.compile(".*invoice_lines.id==(\\S+).*")
+        .matcher(queryParam);
+      final String lineId = lineIdMatcher.find() ? lineIdMatcher.group(1) : EMPTY;
+      
       List<Invoice> invoices;
-      try {
-      invoices = new JsonObject(ApiTestBase.getMockData(MOCK_DATA_INVOICES)).mapTo(InvoiceCollection.class)
-        .getInvoices();
-      } catch (IOException e) {
-        invoices = new ArrayList<Invoice>();
-      }
-      Optional<List<Invoice>> invoiceOptional = getMockEntries(INVOICES, Invoice.class);
-      Invoice invoice0 = invoiceOptional.get()
-        .get(0);
-      invoices.set(0, invoice0);
       InvoiceCollection invoiceCollection = new InvoiceCollection();
-      invoiceCollection.setInvoices(invoices);
-      invoiceCollection.setTotalRecords(invoiceCollection.getInvoices()
-        .size());
-      JsonObject invoicesJson = JsonObject.mapFrom(invoiceCollection);
+      
+      if (lineId.equals(SEARCH_INVOICE_BY_LINE_ID_NOT_FOUND)) {
+
+        invoiceCollection.setInvoices(new ArrayList<Invoice>());
+        invoiceCollection.setTotalRecords(invoiceCollection.getInvoices()
+          .size());
+        JsonObject invoicesJson = JsonObject.mapFrom(invoiceCollection);
+
+        addServerRqRsData(HttpMethod.GET, INVOICES, JsonObject.mapFrom(invoiceCollection));
+        serverResponse(ctx, 404, APPLICATION_JSON, invoicesJson.encode());
```

Comment:
https://s3.amazonaws.com/foliodocs/api/mod-invoice-storage/p/invoice.html#invoice_storage_invoices_get_response
According to the API documentation, `GET /invoice-storage/invoices` cannot return a 404 error.